packages/flutter_tools/test/general.shard/runner/runner_test.dart

```diff
@@ -411,21 +407,23 @@ void main() {
           shutdownHooks: ShutdownHooks(),
         );
 
-        expect((globals.flutterUsage as TestUsage).printedWelcome, false);
+        expect(
+          (globals.logger as BufferLogger).traceText,
+          isNot(contains('Showed analytics consent message.')),
+        );
       },
```

Comment:
I don't immediately see any tests that make sure we actually do show the analytics consent message when we should. I filed and self-assigned https://github.com/flutter/flutter/issues/160374 as a follow-up item.